diff --git a/node_modules/@agoric/cosmic-swingset/lib/ag-solo/init-basedir.js b/node_modules/@agoric/cosmic-swingset/lib/ag-solo/init-basedir.js
index 6a17445..530fdab 100644
--- a/node_modules/@agoric/cosmic-swingset/lib/ag-solo/init-basedir.js
+++ b/node_modules/@agoric/cosmic-swingset/lib/ag-solo/init-basedir.js
@@ -2,6 +2,7 @@ import fs from 'fs';
 import path from 'path';
 import { execFileSync } from 'child_process';
 
+import { assert, details as X } from '@agoric/assert';
 import anylogger from 'anylogger';
 
 const log = anylogger('ag-solo:init');
@@ -20,17 +21,15 @@ export default function initBasedir(
   options.wallet = wallet;
 
   const here = __dirname;
-  try {
-    fs.mkdirSync(basedir, 0o700);
-  } catch (e) {
-    if (!fs.existsSync(path.join(basedir, 'ag-cosmos-helper-address'))) {
-      log.error(
-        `unable to create basedir ${basedir}, it must not already exist`,
-      );
-      throw e;
-    }
+  if (
+    fs.existsSync(basedir) &&
+    !fs.existsSync(path.join(basedir, 'ag-cosmos-helper-address'))
+  ) {
+    assert.fail(X`${basedir} must not already exist`);
   }
 
+  fs.mkdirSync(basedir, { mode: 0o700, recursive: true });
+
   const connections = [{ type: 'http', port: webport, host: webhost }];
   fs.writeFileSync(
     path.join(basedir, 'connections.json'),
diff --git a/node_modules/@agoric/cosmic-swingset/lib/ag-solo/start.js b/node_modules/@agoric/cosmic-swingset/lib/ag-solo/start.js
index 06b4052..5f9dd5e 100644
--- a/node_modules/@agoric/cosmic-swingset/lib/ag-solo/start.js
+++ b/node_modules/@agoric/cosmic-swingset/lib/ag-solo/start.js
@@ -1,7 +1,7 @@
 import fs from 'fs';
 import path from 'path';
 import temp from 'temp';
-import { exec } from 'child_process';
+import { fork } from 'child_process';
 import { promisify } from 'util';
 // import { createHash } from 'crypto';
 
@@ -380,28 +380,35 @@ export default async function start(basedir, argv) {
     .map(dep => path.resolve(agWallet, dep))
     .join(' ');
 
-  const agoricCli = require.resolve('.bin/agoric');
+  const agoricCli = require.resolve('agoric/bin/agoric');
 
   // Use the same verbosity as our caller did for us.
   let verbosity;
   if (process.env.DEBUG === undefined) {
-    verbosity = '';
+    verbosity = [];
   } else if (process.env.DEBUG.includes('agoric')) {
-    verbosity = ' -vv';
+    verbosity = ['-vv'];
   } else {
-    verbosity = ' -v';
+    verbosity = ['-v'];
   }
 
-  // Launch the agoric wallet deploys (if any).
-  const cp = exec(
-    `${agoricCli} deploy${verbosity} --provide=wallet --hostport=${hostport} ${agWalletDeploy}`,
+  // Launch the agoric wallet deploys (if any).  The assumption is that the CLI
+  // runs correctly under the same version of the JS engine we're currently
+  // using.
+  fork(
+    agoricCli,
+    [
+      `deploy`,
+      ...verbosity,
+      `--provide=wallet`,
+      `--hostport=${hostport}`,
+      `${agWalletDeploy}`,
+    ],
+    { stdio: 'inherit' },
     err => {
       if (err) {
         console.error(err);
       }
     },
   );
-
-  cp.stderr.pipe(process.stderr);
-  cp.stdout.pipe(process.stdout);
 }
